src/test/feature/testhawqregister/test-hawq-register.cpp

```diff
@@ -0,0 +1,80 @@
+#include <string>
+
+#include "lib/command.h"
+#include "lib/common.h"
+#include "lib/sql-util.h"
+
+#include "gtest/gtest.h"
+
+using std::string;
+
+/* This test suite may consume more than 80 seconds. */
+class TestHawqRegister : public ::testing::Test {
+ public:
+  TestHawqRegister() {}
+  ~TestHawqRegister() {}
+};
+
+TEST_F(TestHawqRegister, TestSingleFile) {
+	SQLUtility util;
+	string rootPath(util.getTestRootPath()); 
+	string relativePath("/testhawqregister/test_hawq.paq");
+	string filePath = rootPath + relativePath;
+
+	EXPECT_EQ(0, Command::getCommandStatus("hadoop fs -put " + filePath + " /hawq_register_hawq.paq"));
+
+	util.execute("create table hawqregister(i int) with (appendonly=true, orientation=parquet);");
+	util.query("select * from hawqregister;", 0);
+
+	EXPECT_EQ(0, Command::getCommandStatus("hawq register postgres hawqregister /hawq_register_hawq.paq"));
+
+	util.query("select * from hawqregister;", 3);
+	util.execute("insert into hawqregister values(1);");
+	util.query("select * from hawqregister;", 4);
+}
+
+TEST_F(TestHawqRegister, TestSingleHiveFile) {
```

Comment:
And it's also better to add some cases for negative paths. For example, load data into hash table.